opal/Makefile.am

```diff
@@ -53,8 +53,19 @@ lib@OPAL_LIB_PREFIX@open_pal_la_LIBADD = \
         util/libopalutil.la \
 	$(MCA_opal_FRAMEWORK_LIBS)
 lib@OPAL_LIB_PREFIX@open_pal_la_DEPENDENCIES = $(lib@OPAL_LIB_PREFIX@open_pal_la_LIBADD)
+
+if HAVE_ATTRIBUTE_DESTRUCTOR
+
 lib@OPAL_LIB_PREFIX@open_pal_la_LDFLAGS = -version-info $(libopen_pal_so_version)
 
+else
+if HAVE_LD_FINI
+
+lib@OPAL_LIB_PREFIX@open_pal_la_LDFLAGS = -version-info $(libopen_pal_so_version) -Wl,-fini,opal_cleanup
+
+endif
+endif
```

Comment:
I'd change this to:

```
lib@OPAL_LIB_PREFIX@open_pal_la_LDFLAGS = -version-info $(libopen_pal_so_version)
if NEED_LD_FINI
lib@OPAL_LIB_PREFIX@open_pal_la_LDFLAGS += -Wl,fini,opal_cleanup
endif
```

and define the AM_CONDITIONAL for NEED_LD_FINI to be "if linker supports it *and* we the compiler doesn't support the destructor attribute.